wordpress-mobile/WordPress-iOS
# EditPostViewController refactor (1)

Encapsulate all the logic of choosing what editor to launch and instantiating it inside `EditorSettings`.

I've tried to simplify the code in preparation for refactoring the EditPostViewController hierarchy, but also created a new `Restorer` class for restoring simple `UINavigationControllers` to be presented modally. There's no need to have multiple VCs with custom identifiers for that, the app delegate can take care of any VC without a `restorationClass`

The two `instantiate*` methods will get much simpler with Swift 4, but I don't think the workaround is too bad.

```swift
// Swift 4
func instantiatePostEditor(post: AbstractPost) -> UIViewController & PostEditor {
    switch editor {
    case .aztec:
        return AztecPostViewController(post: post)
    case .hybrid:
        return WPPostViewController(post: post)
    case .legacy:
        return WPLegacyEditPostViewController(post: post)
    }
}
```

First part of #7283 

To test:

For all three editors:
- Create a post
- Edit an existing post
- Create a page
- Edit an existing page
- Verify that VC restoration works

Needs review: @nheagy
